planner/core/planbuilder.go

```diff
@@ -910,7 +912,8 @@ func getPossibleAccessPaths(ctx sessionctx.Context, tableHints *tableHintInfo, i
 		// It is syntactically valid to omit index_list for USE INDEX, which means “use no indexes”.
 		// Omitting index_list for FORCE INDEX or IGNORE INDEX is a syntax error.
 		// See https://dev.mysql.com/doc/refman/8.0/en/index-hints.html.
-		if hint.IndexNames == nil && hint.HintType != ast.HintIgnore {
+		if hint.IndexNames == nil && (hint.HintType == ast.HintForce ||
+			(hint.HintType == ast.HintUse && !(tableHints != nil && len(tableHints.tiflashTables) > 0))) {
```

Comment:
We should make `explain format = 'hint'` return correct result, i.e, no `use_index(@`sel_1` `test`.`t` )` in the hint, instead of handling this invalid input hint?

Reply:
Good idea, I will have a try on this solution.

Reply:
A potential difficulty of this way is that lots of tests will be broken.

Reply:
Why? only the result of `explain format = 'hint'` for tiflash plan would change?